portal-api/src/v1/controllers/countries.controller.js

```diff
@@ -1,44 +1,122 @@
 const utils = require('../../utils/array');
-
 const externalApi = require('../../external-api/api');
 
-const getCountryFromArray = (arr, code) => arr.filter((country) => country.code === code)[0];
+/**
+ * Retrieves a country object from an array based on the provided country code.
+ *
+ * @param {Array} arr - The array of country objects to search in.
+ * @param {string} code - The country code to search for.
+ * @returns {Array} - The country object matching the provided code, or an empty array if not found.
+ */
+const getCountryFromArray = (arr, code) => (arr.length ? arr.filter((country) => country.code === code)[0] : []);
 
+/**
+ * Sorts an array of countries.
+ *
+ * @param {Array} countries - The array of countries to be sorted.
+ * @returns {Array} - The sorted array of countries.
+ * @throws {Error} - If unable to sort countries.
+ */
 const sortCountries = (countries) => {
-  const countriesWithoutUK = countries.filter((country) => country.code !== 'GBR');
+  try {
+    const countriesWithoutUK = countries.filter((country) => country.code !== 'GBR');
 
-  const sortedArray = [
-    getCountryFromArray(countries, 'GBR'),
-    ...utils.sortArrayAlphabetically(countriesWithoutUK, 'name'),
-  ];
+    const sortedArray = [getCountryFromArray(countries, 'GBR'), ...utils.sortArrayAlphabetically(countriesWithoutUK, 'name')];
 
-  return sortedArray;
+    return sortedArray;
+  } catch (error) {
+    console.error('Unable to sort countries %o', error);
+    throw new Error('Unable to sort countries', { cause: error });
+  }
 };
 
-const findOneCountry = async (code) => {
-  const response = await externalApi.countries.getCountry(code);
-  return response;
+/**
+ * Retrieves country information from an external API based on the provided country code.
+ *
+ * @param {string} code - The country code.
+ * @returns {Promise<Object>} - A promise that resolves to the country information.
+ * @throws {Error} - If the API response is void or an error occurs while retrieving the country information.
+ */
+const getCountry = async (code) => {
```

Comment:
This does feel like a service-level function as its not dealing with `req` or `res`

Reply:
Happy to tackle this in a separate PR, there other functions which could be updated with similar pattern.